packages/editor/src/components/document-outline/index.js

```diff
@@ -70,17 +70,8 @@ export const DocumentOutline = ( { blocks = [], title, onSelect, isTitleSupporte
 
 	let prevHeadingLevel = 1;
 
-	// Select the corresponding block in the main editor
-	// when clicking on a heading item from the list.
-	const onSelectHeading = ( clientId ) => onSelect( clientId );
-	const focusTitle = () => {
-		// Not great but it's the simplest way to focus the title right now.
-		const titleNode = document.querySelector( '.editor-post-title__input' );
-		if ( titleNode ) {
-			titleNode.focus();
-		}
-	};
-
+	// Not great but it's the simplest way to locate the title right now.
+	const titleNode = document.querySelector( '.editor-post-title__input' );
```

Comment:
Is there no selector to get the title? This seems quite brittle.

I wouldn't advocate for this staying, but if it does it needs an E2E test to make sure changing that selector won't regress things.

Reply:
this was the only way I could 'discover' the selector though I agree it is brittle. Maybe @aduth has another suggestion how we could do that here.

Otherwise I'll try to add an e2e test.

Reply:
re: adding e2e tests for this selector:

Reviewing the e2e testing suite, i see numerous existing instances where we rely on the `.editor-post-title__input` selector ( see https://github.com/WordPress/gutenberg/search?q=.editor-post-title__input&unscoped_q=.editor-post-title__input) - if this selector didn't exist all these tests would fail.

Reply:
I'm generally fine with this in that it's at least what had existed previously. The two downsides I see with the changes proposed here are:

- The selector is now executed at the time of rendering, rather than at the time of click, leaving the potential for some drift to occur.
- It assumes (below) that the node has an `id` associated with it, which did not exist previously.

These are generally minor, and are "par for the course" in updating the implementation to leverage `id` as the anchor point.

Reply:
@aduth @tofumatt are you ok with the current state of this PR?  Are there any additional points I can work to address?

Reply:
One thing I might like to see: The previous implementation guarded against the (real) possibility that the title node could not be found. Here we've removed that protection, which could crash the editor. It could be worth considering to include this in the derived value for `hasTitle`, which would prevent the rendering of the `DocumentOutlineItem` if the title node does not exist.

Reply:
> One thing I might like to see: The previous implementation guarded against the (real) possibility that the title node could not be found.

This was never addressed?